File: RxUI/src/main/java/com/tamsiree/rxui/view/progressing/style/MultiplePulse.java
package com.tamsiree.rxui.view.progressing.style;


import com.tamsiree.rxui.view.progressing.sprite.Sprite;
import com.tamsiree.rxui.view.progressing.sprite.SpriteContainer;

/**
 * @author tamsiree
 */
public class MultiplePulse extends SpriteContainer {
    @Override
    public Sprite[] onCreateChild() {
        return new Sprite[]{
                new Pulse(),
                new Pulse(),
                new Pulse(),
        };
    }

    @Override
    public void onChildCreated(Sprite... sprites) {
        for (int i = 0; i < sprites.length; i++) {
            sprites[i].setAnimationDelay(200 * (i + 1));
        }
    }
}

File: RxUI/src/main/java/com/tamsiree/rxui/view/progressing/style/MultiplePulse.kt
package com.tamsiree.rxui.view.progressing.style

import com.tamsiree.rxui.view.progressing.sprite.Sprite
import com.tamsiree.rxui.view.progressing.sprite.SpriteContainer

/**
 * @author tamsiree
 */
class MultiplePulse : SpriteContainer() {
    override fun onCreateChild(): Array<Sprite?>? {
        return arrayOf(
                Pulse(),
                Pulse(),
                Pulse())
    }

    override fun onChildCreated(vararg sprites: Sprite?) {
        for (i in sprites.indices) {
            sprites[i]?.setAnimationDelay(200 * (i + 1))
        }
    }
}